examples_tests/49.ComputeFFT/fft_convolve_ifft.comp

```diff
@@ -0,0 +1,96 @@
+// WorkGroup Size
+
+#define _NBL_GLSL_WORKGROUP_SIZE_ 256
+
+layout(local_size_x=_NBL_GLSL_WORKGROUP_SIZE_, local_size_y=1, local_size_z=1) in;
+
+ 
+#define _NBL_GLSL_EXT_FFT_GET_PARAMETERS_DEFINED_
+#define _NBL_GLSL_EXT_FFT_GET_DATA_DEFINED_
+#define _NBL_GLSL_EXT_FFT_SET_DATA_DEFINED_
+#define _NBL_GLSL_EXT_FFT_GET_PADDED_DATA_DEFINED_
+#include "nbl/builtin/glsl/ext/FFT/fft.glsl"
+
+// Input Descriptor
+
+layout(set=0, binding=0) readonly restrict buffer InputBuffer
+{
+	nbl_glsl_complex inData[];
+};
+
+layout(set=0, binding=1) restrict readonly buffer KernelBuffer
+{
+	nbl_glsl_complex kerData[];
+};
+
+// Output Descriptor
+
+layout(set=0, binding=2) restrict buffer OutputBuffer
+{
+	nbl_glsl_complex outData[];
+};
+
+// Get/Set Data Function
+layout(push_constant) uniform PushConstants
+{
+	layout (offset = 0) nbl_glsl_ext_FFT_Parameters_t params;
+} pc;
+
+ nbl_glsl_ext_FFT_Parameters_t nbl_glsl_ext_FFT_getParameters() {
+	 nbl_glsl_ext_FFT_Parameters_t ret;
+	 ret = pc.params;
+	 return ret;
+ }
+
+nbl_glsl_complex nbl_glsl_ext_FFT_getData(in uvec3 coordinate, in uint channel)
+{
+	nbl_glsl_complex retValue = nbl_glsl_complex(0, 0);
+	uvec3 dimension = nbl_glsl_ext_FFT_getDimensions();
+	uint index = channel * (dimension.x * dimension.y * dimension.z) + coordinate.z * (dimension.x * dimension.y) + coordinate.y * (dimension.x) + coordinate.x;
+	retValue = inData[index];
+	return retValue;
+}
+
+void nbl_glsl_ext_FFT_setData(in uvec3 coordinate, in uint channel, in nbl_glsl_complex complex_value)
+{
+	uvec3 dimension = nbl_glsl_ext_FFT_getPaddedDimensions();
+	uint index = channel * (dimension.x * dimension.y * dimension.z) + coordinate.z * (dimension.x * dimension.y) + coordinate.y * (dimension.x) + coordinate.x;
+	outData[index] = complex_value;
+}
+
+nbl_glsl_complex nbl_glsl_ext_FFT_getPaddedData(in uvec3 coordinate, in uint channel) {
+
+	uvec3 max_coord = nbl_glsl_ext_FFT_getDimensions() - uvec3(1u);
+	uvec3 clamped_coord = min(coordinate, max_coord);
+	
+	bool is_out_of_range = any(bvec3(coordinate!=clamped_coord));
+
+	uint paddingType = nbl_glsl_ext_FFT_getPaddingType();
+
+	if (_NBL_GLSL_EXT_FFT_FILL_WITH_ZERO_ == paddingType && is_out_of_range) {
+		return nbl_glsl_complex(0, 0);
+	}
+	
+	return nbl_glsl_ext_FFT_getData(clamped_coord, channel);
+}
+
+void convolve() 
+{
+	uint idx = 0;
+    inData[idx] = nbl_glsl_complex_mul(outData[idx], kerData[idx]);
+}
+
+void main()
+{
+	nbl_glsl_ext_FFT(nbl_glsl_ext_FFT_getIsInverse()); // inData->outData
+	
+    barrier();
+    memoryBarrierShared();
+
+	convolve();	// outData+kerData->inData
+	
+    barrier();
+    memoryBarrierShared();
+
+	nbl_glsl_ext_FFT(!nbl_glsl_ext_FFT_getIsInverse()); // inData->outData
```

Comment:
tip, use local variables, declare them in global scope so `getData` and `setData` fetch from the thread-local variables

handle the pre-loading of the local variables and saving in the `main()` function

This is what your local variable could look like.
```glsl
float dataStorage[NBL_GLSL_EXT_FFT_AXIS_LENGTH/_NBL_GLSL_WORKGROUP_SIZE_][NBL_GLSL_EXT_FFT_CHANNEL_COUNT];
```

P.S. the result of `NBL_GLSL_EXT_FFT_AXIS_LENGTH/_NBL_GLSL_WORKGROUP_SIZE_` would need to be rounded up to a power of two.

Reply:
hmm actually for this kernel you could have separate channels in separate threads/workgroups because the input/ouput is an SSBO

Reply:
but I'd make `nbl_glsl_ext_FFT` take a `uint channelIndex` parameter